gix-url/src/lib.rs

```diff
@@ -322,7 +332,13 @@ impl Url {
         }
         match (&self.user, &self.host) {
             (Some(user), Some(host)) => {
-                out.write_all(percent_encode(user).as_bytes())?;
+                // For SCP-like SSH URLs, don't percent-encode underscores in usernames
+                // For regular SSH URLs, use standard encoding but allow underscores
+                let encoded_user = match self.scheme {
+                    Scheme::Ssh => percent_encode_for_scp_user(user),
```

Comment:
I've implemented this separately: https://github.com/GitoxideLabs/gitoxide/pull/2060

I'm not sure which version is better, but I don't think we should encode `@`, etc. because Git doesn't decode these characters.

https://github.com/git/git/blob/v2.50.0/connect.c#L1081

Reply:
Thank you for researching this!